### PR TITLE
Event Reprocessing / Processing Issues

### PR DESCRIPTION
**Pull request by mitsuhiko**
This pull request adds reprocessing support to the system. So far this feature is only enabled for native symbol processing.

## Project Settings UI

screen shot 2017-02-09 at 16 08 02

## Different Stream Details

screen shot 2017-02-09 at 16 05 40

screen shot 2017-02-09 at 16 06 55

screen shot 2017-02-09 at 16 07 11

### CODE REVIEW

**Comment by HazAT** [comment]
screenshot 2017-02-09 14 16 28
screenshot 2017-02-09 14 12 56
screenshot 2017-02-09 13 55 11
screenshot 2017-02-09 14 12 50
*This actually says Issue not Event

**Comment by mitsuhiko** [comment]
The red bar needs a space before the timestamp.

**Comment by HazAT** [comment]
Reminder ... Check timestamps ... The reprocessed events all get the same timestamp, they should keep the timestamp of the event

**Comment by ckj** [comment]
@mitsuhiko @HazAT this is great. what do I see when I click 'show details' when there are issues that block processing?

**Comment by mitsuhiko** [comment]
@ckj it takes you to the projects settings page for processing issues with a list of all processing issues (like which dsyms are missing etc.). There is also a button to trigger reprocessing manually and help about how to resolve it. A partial screenshot of that page is the first link.

Additionally the navigation in the project settings shows a badge with the number of processing issues.

**Comment by HazAT** [comment]
@ckj The missing screenshot ^^
screenshot 2017-02-10 09 04 50

**Comment by HazAT** [comment]
https://facebook.github.io/react/docs/uncontrolled-components.html#default-values
You are right ... If I should change it I would have to initalize the whole processing issues jsx file to receive the form values upfront.

**Comment by mitsuhiko** [comment]
Since this has the change to break other stuff, what if we just make it a button to turn on/off for now?

File: src/sentry/tasks/store.py [hunk 1]
```diff
@@ -112,15 +130,93 @@ def process_event(cache_key, start_time=None, **kwargs):
...
+    """If processing failed we put the original data from the cache into a
+    raw event.
+    """
+    # We need to get the original data here instead of passing the data in
+    # from the last processing step because we do not want any
+    # modifications to take place.
+    delete_raw_event(project, event_id)
+    data = default_cache.get(cache_key)
+    if data is None:
+        metrics.incr('events.failed', tags={'reason': 'cache', 'stage': 'failed_raw'})
```
Comment (JTCunning) [inline src/sentry/tasks/store.py]: `metrics.incr('events.failed', tags={'reason': 'cache', 'stage': 'raw'})`

File: src/sentry/conf/server.py [hunk 2]
```diff
@@ -446,7 +447,12 @@ def env(key, default='', type=None):
     Queue('digests.scheduling', routing_key='digests.scheduling'),
     Queue('email', routing_key='email'),
     Queue('events.preprocess_event', routing_key='events.preprocess_event'),
+    Queue('events.preprocess_event_from_reprocessing',
```
Comment (JTCunning) [inline src/sentry/conf/server.py]: I'd be inclined to rename this queue and the ones below to:
```
events.reprocess.preprocess_event
events.reprocess.process_event
events.reprocess.schedule
```
Seems a little more fitting of description for me.

**Review by JTCunning — COMMENTED** [review]
Added a few things that should probably spark either change or conversation, but nothing blocking.

**Comment by mitsuhiko** [comment]
I think as it stands right now this might be ready to review and then merge.

**Comment by mikepurvis** [comment]
This change is going to be huge for enabling custom crash reporters.

**Comment by HazAT** [comment]
screenshot 2017-02-17 15 18 12

**Comment by HazAT** [comment]
**Do not rebase/squash merge this branch!**
Itunes connect branch builds on it
https://github.com/getsentry/sentry/pull/4574

**Review by JTCunning — APPROVED** [review]
Burn it all down.

**Pull request merged**